Mage.Tests/src/test/java/org/mage/test/cards/continuous/MerfolkTricksterTest.java

```diff
@@ -0,0 +1,158 @@
+    @Test
+    public void test_TricksterAndFlyer_FlyingRemoved()
+    {
+        addCard(Zone.BATTLEFIELD, playerA, "Flying Men"); // (U) 1/1 flyer
+        addCard(Zone.BATTLEFIELD, playerB, "Island", 2);
+        addCard(Zone.HAND, playerB, mTrickster);
+        
+        attack(1, playerA, "Flying Men");
+        castSpell(1, PhaseStep.DECLARE_BLOCKERS, playerB, mTrickster);
+        
+        setStopAt(1, PhaseStep.END_COMBAT);
+        execute();
+        
+        assertLife(playerA, 20);
+        assertLife(playerB, 19);
+        assertTappedCount("Island", true, 2);
+        assertTapped("Flying Men", true);
+        
+        Abilities<Ability> noAbilities = new AbilitiesImpl<>();
+        assertAbilities(playerA, "Flying Men", noAbilities); // no abilities, empty list
+        
+        Abilities<Ability> flashAbility = new AbilitiesImpl<>();
+        flashAbility.add(FlashAbility.getInstance());
+        assertAbilities(playerB, mTrickster, flashAbility); // has flash        
+        
+        assertAllCommandsUsed();
+    }
+    
+    @Test
+    public void test_TricksterAndFlyerBlocked_FlyingRemovedAndBlocked()
+    {
+        addCard(Zone.BATTLEFIELD, playerA, "Flying Men"); // (U) 1/1 flyer
+        addCard(Zone.BATTLEFIELD, playerB, "Island", 2);
+        addCard(Zone.HAND, playerB, mTrickster);
+        
+        attack(1, playerA, "Flying Men");
+        castSpell(1, PhaseStep.DECLARE_ATTACKERS, playerB, mTrickster);
+        block(1, playerB, mTrickster, "Flying Men");
+        
+        setStopAt(1, PhaseStep.END_COMBAT);
+        execute();
+        
+        assertLife(playerA, 20);
+        assertLife(playerB, 20);
+        assertTappedCount("Island", true, 2);
+        assertGraveyardCount(playerA, "Flying Men", 1);
+        assertPermanentCount(playerB, mTrickster, 1);
+        assertDamageReceived(playerB, mTrickster, 1);
+        assertAllCommandsUsed();
+    }
+    
+    @Test
+    public void test_TricksterBlocksFootlightFiend_Survives()
+    {
+        addCard(Zone.BATTLEFIELD, playerA, "Footlight Fiend"); // (R/B) 1/1 on death pings any target for 1
+        addCard(Zone.BATTLEFIELD, playerB, "Island", 2);
+        addCard(Zone.HAND, playerB, mTrickster);
+        
+        attack(1, playerA, "Footlight Fiend");
+        castSpell(1, PhaseStep.DECLARE_ATTACKERS, playerB, mTrickster);
+        block(1, playerB, mTrickster, "Footlight Fiend");
+        addTarget(playerA, mTrickster);
+        
+        setStopAt(1, PhaseStep.END_COMBAT);
+        execute();
+        
+        assertLife(playerA, 20);
+        assertLife(playerB, 20);
+        assertTappedCount("Island", true, 2);
+        assertPermanentCount(playerB, mTrickster, 1);
+        assertDamageReceived(playerB, mTrickster, 1);
+        //assertAllCommandsUsed(); // uncommenting this will force a failure since PlayerA cannot do a command to target Trickster, as expected
+    }
+    
+    // Making sure Footlight Fiend working correctly here first...    
```

Comment:
Perhaps this test case should be in Footlight Fiends file.

Reply:
Fair point. I was just lazy initially! I'll move it outside of here.